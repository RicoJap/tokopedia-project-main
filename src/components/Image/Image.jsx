import React from "react";
import MuiAvatar from "@material-ui/core/Avatar";

/**
 * Image Component
 *
 * @since  16/11/2020
 * @access (public)
 *
 * @type     Functional Component
 * @param {String}  url the url or location of the image
 *
 */

const Image = ({ url, ...attributes }) => {
  return <MuiAvatar data-testid="image" src={url} {...attributes} />;
};

export default Image;
