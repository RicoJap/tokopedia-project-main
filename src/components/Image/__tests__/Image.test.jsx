import "@testing-library/jest-dom";
import React from "react";
import renderer from "react-test-renderer";

import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import Image from "../Image";
import PokeballIcon from "../../../assets/pokeball-big.png";

const MockImage = () => {
  return <Image url={PokeballIcon} />;
};

describe("Snapshot testing", () => {
  const tree = renderer.create(<MockImage />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Image Component renders correctly', () => {
  test('Check if Image component is present in the document', () => {
		render(<MockImage />);
		expect(screen.getByTestId('image')).toBeInTheDocument();
  });
});