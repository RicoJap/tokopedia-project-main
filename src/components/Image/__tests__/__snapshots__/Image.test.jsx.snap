// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle"
  data-testid="image"
>
  <img
    className="MuiAvatar-img"
    src="pokeball-big.png"
  />
</div>
`;
